### PR TITLE
Remove references of deprecated  load IL opcode

### PR DESCRIPTION
**Pull request by wbh123456**
Remove all references to the IL Opcodes in the `load` category listed in #2657.

Issue: #2657
Signed-off-by: Bohao(Aaron) Wang aaronwang0407@gmail.com

### CODE REVIEW

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by wbh123456** [comment]
Passed extended tests. This PR is ready for a review.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -456,7 +456,6 @@ OMR::ARM64::TreeEvaluator::sloadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return commonLoadEvaluator(node, TR::InstOpCode::ldrshimmx, 2, cg);
    }
 
-// also handles cloadi
 TR::Register *
 OMR::ARM64::TreeEvaluator::cloadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```
Comment (andrewcraik) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: Does this need to be renamed - cload is also gone based on the other removes no?

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -3045,37 +3045,6 @@ generateS390CompareAndBranchOpsHelper(TR::Node * node, TR::CodeGenerator * cg, T
...
-   //   c2i        ; refcnt-1, unevaluated
-   //     icload   ; evaluated
-   //   c2i        ; refcnt-1, unevaluated
-   //     icload   ; evaluated
-   //
-   // Try to generate a CR.
-   // FIXME: can't the binary commutative analyser handle this? that's where this should be done
-   //            what about su2i? s2i? mixing of these? bytes? ints? longs?
-   //
-   else if (firstChild->getOpCodeValue()==TR::su2i && firstChild->getRegister()==NULL && firstChild->getReferenceCount()==1 &&
```
Comment (andrewcraik) [inline compiler/z/codegen/OMRTreeEvaluator.cpp]: I'm not seeing an s2i partner for this so what is the logic for eliminating this all together rather than converting it to s type?

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 3]
```diff
@@ -3086,12 +3055,10 @@ generateS390CompareAndBranchOpsHelper(TR::Node * node, TR::CodeGenerator * cg, T
    // FIXME: can't the binary commutative analyser handle this? that's where this should be done
    //
    else if (firstChild->getOpCodeValue()==TR::bu2i && firstChild->getRegister()==NULL && firstChild->getReferenceCount()==1 &&
-            (firstChild->getFirstChild()->getOpCodeValue()==TR::buloadi   ||
-             firstChild->getFirstChild()->getOpCodeValue()==TR::iRegLoad)
+            firstChild->getFirstChild()->getOpCodeValue()==TR::iRegLoad
             && firstChild->getFirstChild()->getRegister() &&
             secondChild->getOpCodeValue()==TR::bu2i && secondChild->getRegister()==NULL && secondChild->getReferenceCount()==1 &&
-            (secondChild->getFirstChild()->getOpCodeValue()==TR::buloadi  ||
```
Comment (andrewcraik) [inline compiler/z/codegen/OMRTreeEvaluator.cpp]: should this be a bloadi instead?

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 4]
```diff
@@ -5984,7 +5949,7 @@ OMR::Z::TreeEvaluator::lloadEvaluator(TR::Node * node, TR::CodeGenerator * cg)
 
 /**
  * sload Evaluator: load short integer
- *   - also handles cload, isload and icload
+ *   - also handles isload and icload
```
Comment (andrewcraik) [inline compiler/z/codegen/OMRTreeEvaluator.cpp]: Is this comment right? isn't cloadi being eliminated?

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (wbh123456) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: Yes, there is no more creations of `cload` IL Opcode. But `cloadEvaluator()` will be removed in future PR.

File: compiler/z/codegen/OMRTreeEvaluator.cpp (hunk 2 above)
Comment (wbh123456) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: Fixed in d89a0b3

File: compiler/z/codegen/OMRTreeEvaluator.cpp (hunk 3 above)
Comment (wbh123456) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: Fixed in d89a0b3

File: compiler/z/codegen/OMRTreeEvaluator.cpp (hunk 4 above)
Comment (wbh123456) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: Fixed in d89a0b3

**Comment by wbh123456** [comment]
@andrewcraik This PR is ready for a review.

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by wbh123456** [comment]
@andrewcraik @fjeremic Ping

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 5]
```diff
@@ -6229,10 +6227,7 @@ OMR::Z::TreeEvaluator::bstoreEvaluator(TR::Node * node, TR::CodeGenerator * cg)
...
    //
    else if (valueChild->getReferenceCount() == 1 && valueChild->getRegister() == NULL &&
-             (valueChild->getOpCodeValue() == TR::buload  ||
-              valueChild->getOpCodeValue() == TR::bload   ||
-              valueChild->getOpCodeValue() == TR::buloadi ||
-              valueChild->getOpCodeValue() == TR::buload      ))
+             valueChild->getOpCodeValue() == TR::bload   )
       {
       sourceRegister = cg->allocateRegister();
       TR::MemoryReference * tempMR2 = generateS390MemoryReference(valueChild, cg);
```
Comment (fjeremic) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:6233]: I think this optimization should work for `bloadi` as well. However this is outside the scope of this PR. I've opened up to investigate.

**Comment by 0xdaryl** [comment]
This shouldn't merge until @andrewcraik signs off on his requested changes.

File: compiler/z/codegen/OMRTreeEvaluator.cpp (hunk 5 above)
Comment (andrewcraik) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:6233]: not sure it is beyond the scope because the buloadi case would convert to bloadi no?

Comment (fjeremic) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:6233]: I think it _should_, I was just being cautious as it would be enabling a path which would not have been taken today for `bloadi`.

Comment (wbh123456) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: This was a mistake while I tried to map the `buloadi`/`buload` to their signed ILOpcodes. `buloadi` should be converted to `bloadi`. I have fixed this in commit 9e5a25a

**Comment by wbh123456** [comment]
@andrewcraik Ping.

File: compiler/p/codegen/FPTreeEvaluator.cpp [hunk 6]
```diff
@@ -1460,9 +1460,15 @@ TR::Register *OMR::Power::TreeEvaluator::i2fEvaluator(TR::Node *node, TR::CodeGe
    TR::Register *tempReg;
    TR::Register *trgReg;
 
+<<<<<<< HEAD
    if (((cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P7) &&
        (node->getOpCodeValue() == TR::iu2f && (child->getOpCodeValue() == TR::iuload || child->getOpCodeValue() == TR::iuloadi))) ||
        (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P6) &&
+=======
```
Comment (andrewcraik) [inline compiler/p/codegen/FPTreeEvaluator.cpp:1467]: I'm not sure these hack markers should be here...

**Comment by andrewcraik** [comment]
Apart from the hack markers in the file for the conflict that have not been properly resolved I am now fine with these changes.

File: compiler/p/codegen/FPTreeEvaluator.cpp (hunk 6 above)
Comment (wbh123456) [reply compiler/p/codegen/FPTreeEvaluator.cpp]: Sorry for the rebasing error. I have fixed this problem in the new commit.

**Comment by wbh123456** [comment]
@0xdaryl  This PR is ready.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build aix,plinux,x32linux,win

**Comment by 0xdaryl** [comment]
@genie-omr build aix,plinux,x32linux,win

**Comment by 0xdaryl** [comment]
@genie-omr build aix,plinux,x32linux,win

**Comment by 0xdaryl** [comment]
@genie-omr build aix,plinux,x32linux,win
